### PR TITLE
Logging

### PR DESCRIPTION
**Pull request by gauntface**
R: @jeffposnick @addyosmani 

This introduces the following possible ways of logging:

```javascript

logHelper.log('Hello World 1');
logHelper.warn('Hello World 1');
logHelper.debug('Hello World 1');
logHelper.error('Hello World 1');


logHelper.log({
    message: 'Hello World 2'
});
// Same for .debug, .warn, .error

logHelper.log({
    message: 'Hello World 3',
    data: {
        example: 'sw-helpers 4 life',
        this: 'Will print out as key value pairs under Extra Information'
    }
});
// Same for .debug, .warn, .error

logHelper.log({
    message: 'Hello World 4',
    error: new Error('This is printed at top level')
});
// Same for .debug, .warn, .error
```

chrome logs
firefox logs

Edge doesn't support color in console AND it's group support is broken (everything ends up out of sync / out of order - so indentation is a mess).

I've dropped the grouping on edge and fallen back to just plain logs.

However, this looks really spammy. Compare Chrome and Edge for logs with (Message, Error and Data).

error in chrome

edge error

### CODE REVIEW

File: gulp-tasks/build.js [hunk 1]
```diff
@@ -43,6 +45,28 @@ const buildPackage = (projectPath) => {
...
 };
 
+gulp.task('build:shared', () => {
+  return Promise.all([
+    buildJSBundle({
+      rollupConfig: {
+        entry: path.join(__dirname, '..', 'lib', 'log-helper.js'),
+        format: 'umd',
+        moduleName: 'goog.logHelper',
+        plugins: [
```
Comment (jeffposnick) [inline gulp-tasks/build.js]: Are these plugins needed? I don't see any external `require()`s in `log-helper.js`.

File: gulp-tasks/build.js [hunk 2]
```diff
@@ -43,6 +45,28 @@ const buildPackage = (projectPath) => {
     });
 };
 
+gulp.task('build:shared', () => {
+  return Promise.all([
```
Comment (jeffposnick) [inline gulp-tasks/build.js]: There's only one promise, so you can ditch the `Promise.all()`.

File: gulp-tasks/build.js [hunk 3]
```diff
@@ -43,6 +45,28 @@ const buildPackage = (projectPath) => {
     });
 };
 
+gulp.task('build:shared', () => {
```
Comment (jeffposnick) [inline gulp-tasks/build.js]: Just to confirm: this `umd` bundle is being consumed by the test, and not anything else, right?

Comment (gauntface) [inline gulp-tasks/build.js]: yes just the test

**Pull request merged**